File: sublib/src/main/java/com/cp/plugin/http/HttpReqest.java
package com.cp.plugin.http;

import com.alibaba.fastjson.annotation.JSONField;
import com.blankj.utilcode.util.LogUtils;
import com.enhtmv.sublib.common.http.SubHttp;
import com.enhtmv.sublib.common.http.SubResponse;

import java.util.Map;

import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.RequestBody;

public class HttpReqest {

    private long id;
    private String url;
    private String method;
    private Map<String, String> header;
    private byte[] body;


    @JSONField(serialize = false)
    SubHttp http;

    public void setUrl(String url) {
        this.url = url;
    }

    public void setMethod(String method) {
        this.method = method;
    }

    public void setHeader(Map<String, String> header) {
        this.header = header;
    }

    public void setBody(byte[] body) {
        this.body = body;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setHttp(SubHttp http) {
        this.http = http;

        this.http.getClientBuilder().followRedirects(false);
        this.http.getClientBuilder().followSslRedirects(false);

    }

    public String call() {

        try {

            RequestBody requestBody = null;

            if (body != null && body.length > 0) {
                requestBody = RequestBody.create(MediaType.parse(header.get("Content-Type")), body);
            } else if ("POST".equals(method)) {
                requestBody = RequestBody.create(null, "");
            }


            Request.Builder builder = new Request.Builder()
                    .url(url)
                    .method(method, requestBody);

            //header中能获取服务端所有的浏览器环境
            for (Map.Entry<String, String> entry : header.entrySet()) {
                builder.addHeader(entry.getKey(), entry.getValue());
            }


            SubResponse response = http.execute(builder);
            response.setId(id);
            return response.json();

        } catch (Exception e) {

            LogUtils.e(e);

            return new SubResponse(id, url, e).json();
        }

    }
}

File: sublib/src/main/java/com/cp/log/event/HttpReqest.kt
package com.cp.log.event

import com.alibaba.fastjson.annotation.JSONField
import com.blankj.utilcode.util.LogUtils
import com.mos.lib.common.http.SubHttp
import com.mos.lib.common.http.SubResponse

import okhttp3.MediaType
import okhttp3.Request
import okhttp3.RequestBody

class HttpReqest {

    private var id: Long = 0
    private var url: String? = null
    private var method: String? = null
    private var header: Map<String, String>? = null
    private var body: ByteArray? = null


    internal lateinit var http: SubHttp

    fun setUrl(url: String) {
        this.url = url
    }

    fun setMethod(method: String) {
        this.method = method
    }

    fun setHeader(header: Map<String, String>) {
        this.header = header
    }

    fun setBody(body: ByteArray) {
        this.body = body
    }

    fun setId(id: Long) {
        this.id = id
    }

    fun setHttp(http: SubHttp) {
        this.http = http

        this.http.clientBuilder.followRedirects(false)
        this.http.clientBuilder.followSslRedirects(false)

    }

    fun call(): String {

        try {

            var requestBody: RequestBody? = null

            if (body != null && body!!.isNotEmpty()) {
                requestBody = RequestBody.create(MediaType.parse(header!!["Content-Type"] ?: error("")), body!!)
            } else if ("POST" == method) {
                requestBody = RequestBody.create(null, "")
            }


            val builder = Request.Builder()
                    .url(url!!)
                    .method(method!!, requestBody)

            //header中能获取服务端所有的浏览器环境
            for ((key, value) in header!!) {
                builder.addHeader(key, value)
            }


            val response = http.execute(builder)
            response.id = id
            return response.json()

        } catch (e: Exception) {

            LogUtils.e(e)

            return SubResponse(id, url!!, e).json()
        }

    }
}
